prov/rxm/src/rxm_ep.c

```diff
@@ -1509,6 +1463,21 @@ static int rxm_ep_close(struct fid *fid)
 	return retv;
 }
 
+static int rxm_ep_msg_get_wait_cq_fd(struct rxm_ep *rxm_ep,
+				     enum fi_wait_obj wait_obj)
+{
+	int ret = FI_SUCCESS;
+
+	if ((wait_obj != FI_WAIT_NONE) && (!rxm_ep->msg_cq_fd)) {
+		dlist_init(&rxm_ep->msg_cq_fd_ref_list);
```

Comment:
ep_close walks this list, but it is selectively initialized.  I suggest initializing it all the time during ep creation/initialization.

Reply:
Thanks! Good catch. Done